src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -228,26 +240,47 @@ async def run_python_test(
     test_options: TestOptions,
 ) -> TestResult:
     """Runs pytest for one target."""
-    env = {"PYTEST_ADDOPTS": f"--color={'yes' if global_options.options.colors else 'no'}"}
+    add_opts = [f"--color={'yes' if global_options.options.colors else 'no'}"]
+    test_results_file = f"{config.address.path_safe_spec}.xml"
```

Comment:
This can be moved into the `if` statement because it's only needed when `xml_dir` is defined.